lib/global-styles.php

```diff
@@ -291,90 +291,117 @@ function gutenberg_experimental_global_styles_get_core() {
 /**
  * Returns the theme presets registered via add_theme_support, if any.
  *
+ * @param array $settings Existing editor settings.
+ *
  * @return array Config that adheres to the theme.json schema.
  */
-function gutenberg_experimental_global_styles_get_theme_support_settings() {
+function gutenberg_experimental_global_styles_get_theme_support_settings( $settings ) {
 	$theme_settings                       = array();
 	$theme_settings['global']             = array();
 	$theme_settings['global']['settings'] = array();
 
 	// Deprecated theme supports.
-	if ( get_theme_support( 'disable-custom-colors' ) ) {
+	if (
+		array_key_exists( 'disableCustomColors', $settings ) &&
+		$settings['disableCustomColors']
+	) {
```

Comment:
Here and in other checks, it can be simplified to:
```suggestion
	if ( ! empty( $settings['disableCustomColors'] ) ) {
```
`isset` and `empty` are language constructs and are always faster than `array_key_exists` which is a regular function. In addition to that, language constructs are able to handle all missing properties (including any level of nesting) without logging any notices (or errors).

Reply:
Nice, thanks for the tip! I ended up using isset.